clients/src/main/java/org/apache/kafka/common/network/ServerConnectionId.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.common.network;
+
+import java.net.Socket;
+import java.util.Optional;
+
+/**
+ * ServerConnectionId is used to uniquely identify a connection between a client and a server.
+ */
+public class ServerConnectionId {
```

Comment:
If we keep this class in the client module, it's better to give it a more accurate name. Perhaps ConnectionIdOnServer?

Reply:
Hmmm, I somehow felt `ServerConnectionId` is better as it prefixes with `server` itself. I can write the class description appropraitely. If you still think it's not appropriate then I ll move it to `ConnectionIdOnServer`. Please let me know.

Reply:
`ServerConnectionId` needs to remain in the clients module since it's also used by the clients' test code. Perhaps we could rewrite `NioEchoServer` to use a different connection ID scheme to decouple it. In #16460, we wanted to add a more meaningful string, but that's not necessary for aligning with the core server code.

Reply:
@apoorvmittal10 what do you think abort moving `ServerConnectionId` to server module?

Reply:
@chia7712 That we can't do, as we discussed that here: https://github.com/apache/kafka/pull/17519#discussion_r1805238289